netbout-web/src/main/java/com/netbout/email/EmCatch.java

```diff
@@ -0,0 +1,186 @@
+@EqualsAndHashCode(of = { "action", "user", "password" })
+final class EmCatch {
+    /**
+     * Max sleep time of thread.
+     */
+    private static final long MAX_SLEEP_TIME = 24 * 60 * 60 * 1000;
+    /**
+     * Action.
+     */
+    private final transient Action action;
+    /**
+     * Email Login key.
+     */
+    private final transient String user;
+    /**
+     * Email password.
+     */
+    private final transient String password;
+    /**
+     * Email server host.
+     */
+    private final transient String host;
+    /**
+     * Email server port.
+     */
+    private final transient int port;
+    /**
+     * Ctor.
+     * @param act Email message handler
+     * @param usr Email login key
+     * @param pass Email password
+     * @param hst Email server host
+     * @param prt Email server port
+     * @param prd Email server check period in milliseconds
+     * @checkstyle ParameterNumberCheck (3 lines)
+     */
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    EmCatch(final Action act, final String usr, final String pass,
+            final String hst, final int prt, final long prd) {
+        this.action = act;
+        this.user = usr;
+        this.password = pass;
+        this.host = hst;
+        this.port = prt;
+        final Thread monitor = new Thread(
+            new Runnable() {
+                @Override
+                public void run() {
+                    long period = prd;
+                    while (true) {
+                        try {
+                            EmCatch.this.check();
+                            TimeUnit.SECONDS.sleep(period);
+                            period = prd;
+                        // @checkstyle LineLength (1 lines)
+                        } catch (final InterruptedException | MessagingException ex) {
+                            Logger.error(this, "%[exception]s", ex);
+                            if (period < MAX_SLEEP_TIME) {
+                                period = period * 2;
+                            }
+                        }
+                    }
+                }
+            }
+        );
+        monitor.setDaemon(true);
+        monitor.start();
+    }
+    /**
+     * Read unread mail and persist new bout message.
+     * @throws MessagingException If fails
+     */
+    private void check() throws MessagingException {
+        Store store = null;
+        Folder inbox = null;
+        try {
+            final Properties props = new Properties();
+            final Session session = Session.getInstance(props);
+            final URLName url = new URLName(
+                "pop3",
+                this.host,
+                this.port,
+                null,
+                this.user,
+                this.password
+            );
+            store = session.getStore(url);
+            store.connect();
+            inbox = store.getFolder("inbox");
+            inbox.open(Folder.READ_WRITE);
```

Comment:
@erimerturk here we should check `folder.exists()`
see http://docs.oracle.com/javaee/6/api/javax/mail/Store.html#getFolder(java.lang.String)